api/promoter.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/SkynetLabs/skynet-accounts/database"
+	"github.com/julienschmidt/httprouter"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+type (
+	// PromoterSetTierPOST describes the body of a POST request that sets the
+	// user's tier.
+	PromoterSetTierPOST struct {
+		Tier int `json:"tier"`
+	}
+)
+
+// _promoterSetTierPOST sets the given user's tier.
+func (api *API) _promoterSetTierPOST(_ *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub := ps.ByName("sub")
+	var body PromoterSetTierPOST
+	err := parseRequestBodyJSON(req.Body, LimitBodySizeLarge, &body)
+	if err != nil {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	if body.Tier < database.TierFree || body.Tier >= database.TierMaxReserved {
+		api.WriteError(w, fmt.Errorf("invalid tier %d", body.Tier), http.StatusBadRequest)
+		return
+	}
+	ctx := req.Context()
+	u, err := api.staticDB.UserBySub(ctx, sub)
+	if errors.Contains(err, database.ErrUserNotFound) {
+		api.WriteError(w, err, http.StatusNotFound)
+		return
+	}
+	err = api.staticDB.UserSetTier(ctx, u, body.Tier)
```

Comment:
I noticed that this upserts a user. Is that intended behaviour? Might lead to a race if the user is deleted between here and `UserBySub`.

Also we need to guarantee that we can't update tiers for users which are linked to an active stripe subscription.

Reply:
Good catch. No, this wasn't intended and it should not happen. I made that behaviour controllable via a parameter.